storage/src/test/java/org/apache/kafka/tiered/storage/TieredStorageTestContext.java

```diff
@@ -0,0 +1,322 @@
+                Collections.singletonMap(configResource, alterEntries);
+        admin.incrementalAlterConfigs(configsMap, alterOptions).all().get(30, TimeUnit.SECONDS);
+    }
+
+    public void deleteTopic(String topic) {
+        TestUtils.deleteTopicWithAdmin(admin, topic, harness.brokers());
+    }
+
+    /**
+     * Send the given records trying to honor the batch size. This is attempted
+     * with a large producer linger and the use of an explicit flush every time
+     * the number of a "group" of records reaches the batch size.
+     * @param recordsToProduce the records to produce
+     * @param batchSize the batch size
+     */
+    public void produce(List<ProducerRecord<String, String>> recordsToProduce, Integer batchSize) {
+        int counter = 0;
+        for (ProducerRecord<String, String> record : recordsToProduce) {
+            producer.send(record);
+            if (counter++ % batchSize == 0) {
+                producer.flush();
+            }
+        }
+    }
+
+    public List<ConsumerRecord<String, String>> consume(TopicPartition topicPartition,
+                                                        Integer expectedTotalCount,
+                                                        Long fetchOffset) {
+        consumer.assign(Collections.singletonList(topicPartition));
+        consumer.seek(topicPartition, fetchOffset);
+
+        long timeoutMs = 60_000L;
+        String sep = System.lineSeparator();
+        List<ConsumerRecord<String, String>> records = new ArrayList<>();
+        Function1<ConsumerRecords<String, String>, Object> pollAction = polledRecords -> {
+            polledRecords.forEach(records::add);
+            return records.size() >= expectedTotalCount;
+        };
+        Function0<String> messageSupplier = () ->
+                String.format("Could not consume %d records of %s from offset %d in %d ms. %d message(s) consumed:%s%s",
+                        expectedTotalCount, topicPartition, fetchOffset, timeoutMs, records.size(), sep,
+                        Utils.join(records, sep));
+        TestUtils.pollRecordsUntilTrue(consumer, pollAction, messageSupplier, timeoutMs);
+        return records;
+    }
+
+    public Long nextOffset(TopicPartition topicPartition) {
+        List<TopicPartition> partitions = Collections.singletonList(topicPartition);
+        consumer.assign(partitions);
+        consumer.seekToEnd(partitions);
+        return consumer.position(topicPartition);
+    }
+
+    public Long beginOffset(TopicPartition topicPartition) {
+        List<TopicPartition> partitions = Collections.singletonList(topicPartition);
+        consumer.assign(partitions);
+        consumer.seekToBeginning(partitions);
+        return consumer.position(topicPartition);
+    }
+
+    public void bounce(int brokerId) {
+        closeClients();
+        harness.killBroker(brokerId);
+        harness.startBroker(brokerId);
+        initContext();
```

Comment:
why do we create new clients every time on bounce? The clients should be able to recover and reconnect.